### PR TITLE
V1 Complete

### PR DESCRIPTION
**Pull request by manas-raj-shrestha**
Please review my code. @amannepid @rajeshkumarkhadka

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -25,4 +25,16 @@ dependencies {
     compile 'com.android.support:cardview-v7:21.0.+'
     compile 'com.android.support:recyclerview-v7:21.0.+'
     compile 'de.hdodenhof:circleimageview:1.2.2'
+    compile 'com.android.support:palette-v7:21.0.0'
```
Comment (amannepid) [inline app/build.gradle:28]: @leapfrog-manas You can use support versions 22.x.x

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.example.manas.movieapp" >
-
+<uses-permission android:name="android.permission.INTERNET"></uses-permission>
```
Comment (amannepid) [inline app/src/main/AndroidManifest.xml]: "android.permission.INTERNET"/> less code ;)

File: app/src/main/java/com/example/manas/movieapp/ActorProfile.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import retrofit.RestAdapter;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+/**
+ * Created by Manas on 3/30/2015.
+ */
+public class ActorProfile extends ActionBarActivity {
+    Toolbar toolbar;
+    String baseurl = "http://api.themoviedb.org/3";
```
Comment (amannepid) [inline app/src/main/java/com/example/manas/movieapp/ActorProfile.java:32]: move these kinds of codes to utils.Constants

File: app/src/main/java/com/example/manas/movieapp/Info/SinlgeMovie.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    public String vote_count;
+    public Reviews reviews;
+    public Casts casts;
+
+    public class Casts {
+        public List<Cast> cast;
+    }
+
+    public class Cast {
+        public String cast_id;
```
Comment (amannepid) [inline app/src/main/java/com/example/manas/movieapp/Info/SinlgeMovie.java:40]: you can create variable castId follwoing Java Naming Convention and use synchronize name instead.

File: app/src/main/java/com/example/manas/movieapp/fragments/NavigationDrawerFragmentHandler.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+                mPosition = position; //put value of currently selected item to mPosition.
+
+                SharedPreferences.Editor edit = movie.edit();
+                edit.putInt("position", position);
+                edit.commit();
+
+                switch (position) {
+                    case 0:
+                        fragmentTransaction.replace(R.id.fragment_for_homepage, new MostPopularMovies());
+                        fragmentTransaction.commit();
```
Comment (khadkarajesh) [inline app/src/main/java/com/example/manas/movieapp/fragments/NavigationDrawerFragmentHandler.java]: don't write same code in multilple place.

File: app/src/main/java/com/example/manas/movieapp/Adapters/CastAdapter.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        TextView casttv = (TextView) v.findViewById(R.id.castTV);
+        casttv.setText(sinlgeMovie.casts.cast.get(position).name +"\nAs \n" + sinlgeMovie.casts.cast.get(position).character);
+        Picasso.with(context).load("http://image.tmdb.org/t/p/w500"+sinlgeMovie.casts.cast.get(position).profile_path)
+               .error(R.drawable.actorpicdefault).placeholder(R.drawable.actorpicdefault) .into(castiv);
+        v.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Bundle bundle = new Bundle();
+                bundle.putString("id",sinlgeMovie.casts.cast.get(position).id);
+                Intent i = new Intent(context, ActorProfile.class);
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/example/manas/movieapp/Adapters/CastAdapter.java:60]: use a interface instead.

File: app/src/main/java/com/example/manas/movieapp/Adapters/HomeFragmentRCVAdapter.java [hunk 7]
```diff
@@ -33,91 +25,82 @@
...
             Bundle bundle = new Bundle();
-            bundle.putString("name", name);
-            bundle.putString("rating", rating);
-            bundle.putString("synopsis", synopsis);
-            bundle.putInt("poster_id", poster_id);
+            bundle.putString("id", id);
 
-            SingleMovieFragmentHandler singleMovieFragmentHandler = new SingleMovieFragmentHandler(bundle, fragmentManager);
-            fragmentTransaction.setCustomAnimations(android.R.anim.slide_in_left,0,android.R.anim.slide_out_right,0);
+            mainActivity.startSingleMovieViewActivity(bundle, viewGroup);
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/example/manas/movieapp/Adapters/HomeFragmentRCVAdapter.java:118]: use a interface instead

**Pull request closed without merging**